src/scaffoldmaker/meshtypes/meshtype_3d_stellate1.py

```diff
@@ -384,16 +490,65 @@ def refineMesh(cls, meshrefinement, options):
         meshrefinement.refineAllElementsCubeStandard3d(refineElementsCount1, refineElementsCount2, refineElementsCount3)
 
 
-def createArm(halfArmArcAngleRadians, elementLengths, armAngle, armAngleConst, elementsCount, dipMultiplier, armCount, armIndex):
+    @classmethod
+    def defineFaceAnnotations(cls, region, options, annotationGroups):
+        """
+        Add point annotation groups from the 1D mesh.
+        :param region: Zinc region containing model.
+        :param options: Dict containing options. See getDefaultOptions().
+        :param annotationGroups: List of annotation groups for top-level elements.
+        New point annotation groups are appended to this list.
+        """
+        # create  groups
+        fm = region.getFieldmodule()
+        armCount = len(options['Numbers of elements along arms'])
+        stellateGroup = getAnnotationGroupForTerm(annotationGroups, get_stellate_term("cervicothoracic ganglion"))
+        armNames, faceNames = getUnlabelledStellateArmNames(armCount)
+        arm1Group = getAnnotationGroupForTerm(annotationGroups, armNames[0])
+        arm2Group = getAnnotationGroupForTerm(annotationGroups, armNames[1])
+        arm3Group = getAnnotationGroupForTerm(annotationGroups, armNames[2])
+
+        mesh2d = fm.findMeshByDimension(2)
+        is_exterior = fm.createFieldIsExterior()
+        is_exterior_face_xi2_0 = fm.createFieldAnd(is_exterior, fm.createFieldIsOnFace(Element.FACE_TYPE_XI2_0))
+        is_exterior_face_xi2_1 = fm.createFieldAnd(is_exterior, fm.createFieldIsOnFace(Element.FACE_TYPE_XI2_1))
+
+        is_arm1 = arm1Group.getFieldElementGroup(mesh2d)
+        is_arm2 = arm2Group.getFieldElementGroup(mesh2d)
+        is_arm3 = arm3Group.getFieldElementGroup(mesh2d)
+        is_left_face = fm.createFieldOr(
+            fm.createFieldAnd(is_arm2, is_exterior_face_xi2_1),
+            fm.createFieldAnd(is_arm3, is_exterior_face_xi2_0))
+        leftStellateGroup = findOrCreateAnnotationGroupForTerm(annotationGroups, region, faceNames[1])
+        leftStellateGroup.getMeshGroup(mesh2d).addElementsConditional(is_left_face)
+
+        is_top_face = fm.createFieldOr(
+            fm.createFieldAnd(is_arm1, is_exterior_face_xi2_1),
+            fm.createFieldAnd(is_arm2, is_exterior_face_xi2_0))
+        topStellateGroup = findOrCreateAnnotationGroupForTerm(annotationGroups, region, faceNames[0])
+        topStellateGroup.getMeshGroup(mesh2d).addElementsConditional(is_top_face)
+
+        is_bottom_face = fm.createFieldOr(
+            fm.createFieldAnd(is_arm1, is_exterior_face_xi2_0),
+            fm.createFieldAnd(is_arm3, is_exterior_face_xi2_1))
+        bottomStellateGroup = findOrCreateAnnotationGroupForTerm(annotationGroups, region, faceNames[-1])
+        bottomStellateGroup.getMeshGroup(mesh2d).addElementsConditional(is_bottom_face)
```

Comment:
You should be able to make a loop over arms and faces that would work with any number of arms AND be less code than this. The mesh2d, is_exterior etc. lines can go before loop, then (using the clever python trick of negative array indexes counting back from the end of the list):
```
armGroups = [getAnnotationGroupForTerm(annotationGroups, armTerm]) for armTerm in armTerms]
isArm = [armGroup.getFieldElementGroup(armGroup) for armGroup in armGroups]
for arm in armCount:
    is_face = fm.createFieldOr(
            fm.createFieldAnd(is_arm[arm - 1], is_exterior_face_xi2_1),
            fm.createFieldAnd(is_arm[arm], is_exterior_face_xi2_0))
    faceGroup = findOrCreateAnnotationGroupForTerm(annotationGroups, region, faceTerms[arm - 1])
    faceGroup.getMeshGroup(mesh2d).addElementsConditional(is_face)
```
DO check my logic.